From 33efd3c5e31081209d6d3b3bfc28b4dfb3588f02 Mon Sep 17 00:00:00 2001
From: Valentin Raevsky <valentin@compulab.co.il>
Date: Tue, 4 Dec 2018 15:32:10 +0200
Subject: [PATCH 100/103] cl-som-imx8: dts: Add usb_mux regulator

Add usb_mux regulator. The regulator controls the SoM
usb_mux. This mux switches between the Intel bluetooth &
the usb-hub. Available values: low: usb-hub; high: bluetooth.

Signed-off-by: Valentin Raevsky <valentin@compulab.co.il>
---
 arch/arm64/boot/dts/compulab/Makefile              |  1 +
 arch/arm64/boot/dts/compulab/cl-som-imx8.dts       |  1 -
 .../arm64/boot/dts/compulab/sbc-imx8-bluetooth.dts | 26 ++++++++++++++++++++++
 arch/arm64/boot/dts/compulab/sbc-imx8.dts          |  5 +++++
 4 files changed, 32 insertions(+), 1 deletion(-)
 create mode 100644 arch/arm64/boot/dts/compulab/sbc-imx8-bluetooth.dts

diff --git a/arch/arm64/boot/dts/compulab/Makefile b/arch/arm64/boot/dts/compulab/Makefile
index 51ca4fb..eaf1b29 100644
--- a/arch/arm64/boot/dts/compulab/Makefile
+++ b/arch/arm64/boot/dts/compulab/Makefile
@@ -1,5 +1,6 @@
 dtb-$(CONFIG_ARCH_FSL_IMX8MQ) += cl-som-imx8.dtb
 dtb-$(CONFIG_ARCH_FSL_IMX8MQ) += sbc-imx8.dtb
+dtb-$(CONFIG_ARCH_FSL_IMX8MQ) += sbc-imx8-bluetooth.dtb
 dtb-$(CONFIG_ARCH_FSL_IMX8MQ) += sbc-imx8-lvds-dcss.dtb
 dtb-$(CONFIG_ARCH_FSL_IMX8MQ) += sbc-imx8-lvds-lcdif.dtb
 dtb-$(CONFIG_ARCH_FSL_IMX8MQ) += ucm-imx8.dtb
diff --git a/arch/arm64/boot/dts/compulab/cl-som-imx8.dts b/arch/arm64/boot/dts/compulab/cl-som-imx8.dts
index 5163d2d..80abe3e 100644
--- a/arch/arm64/boot/dts/compulab/cl-som-imx8.dts
+++ b/arch/arm64/boot/dts/compulab/cl-som-imx8.dts
@@ -36,7 +36,6 @@
 			regulator-max-microvolt = <3300000>;
 			gpio = <&gpio1 4 GPIO_ACTIVE_HIGH>;
 			regulator-always-on;
-			enable-active-high;
 		};
 	};
 };
diff --git a/arch/arm64/boot/dts/compulab/sbc-imx8-bluetooth.dts b/arch/arm64/boot/dts/compulab/sbc-imx8-bluetooth.dts
new file mode 100644
index 0000000..468aa39
--- /dev/null
+++ b/arch/arm64/boot/dts/compulab/sbc-imx8-bluetooth.dts
@@ -0,0 +1,26 @@
+/*
+ * Copyright (C) 2018 CopuLab Ltd.
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+
+#include "cl-som-imx8.dts"
+#include "sb-imx8.dtsi"
+
+/ {
+    model = "CompuLab CL-SOM-iMX8 rev1.1 on SBC-iMX8";
+    compatible = "compulab,sbc-imx8", "compulab,cl-som-imx8,rev1.1", "compulab,cl-som-imx8", "fsl,imx8mq";
+};
+
+/* bluetooth is on */
+&reg_usb_mux {
+    enable-active-high;
+};
diff --git a/arch/arm64/boot/dts/compulab/sbc-imx8.dts b/arch/arm64/boot/dts/compulab/sbc-imx8.dts
index 6ce425f..51a2622 100644
--- a/arch/arm64/boot/dts/compulab/sbc-imx8.dts
+++ b/arch/arm64/boot/dts/compulab/sbc-imx8.dts
@@ -19,3 +19,8 @@
     model = "CompuLab CL-SOM-iMX8 rev1.1 on SBC-iMX8";
     compatible = "compulab,sbc-imx8", "compulab,cl-som-imx8,rev1.1", "compulab,cl-som-imx8", "fsl,imx8mq";
 };
+
+/* usb hub is on */
+&reg_usb_mux {
+    enable-active-low;
+};
-- 
1.9.1

